drivers/timer/ambiq_stimer.c

```diff
@@ -182,25 +194,27 @@ static int stimer_init(void)
 {
 	uint32_t oldCfg;
 
-	oldCfg = am_hal_stimer_config(AM_HAL_STIMER_CFG_FREEZE);
+	oldCfg = am_hal_stimer_config(TIMER_CLKSRC | AM_HAL_STIMER_CFG_FREEZE);
 
 #if defined(CONFIG_SOC_SERIES_APOLLO3X)
 	am_hal_stimer_config((oldCfg & ~(AM_HAL_STIMER_CFG_FREEZE | CTIMER_STCFG_CLKSEL_Msk)) |
-			     TIMER_CLKSRC | AM_HAL_STIMER_CFG_COMPARE_A_ENABLE);
+			     TIMER_CLKSRC | AM_HAL_STIMER_CFG_COMPARE_A_ENABLE |
+			     AM_HAL_STIMER_CFG_COMPARE_B_ENABLE);
 #else
 	am_hal_stimer_config((oldCfg & ~(AM_HAL_STIMER_CFG_FREEZE | STIMER_STCFG_CLKSEL_Msk)) |
-			     TIMER_CLKSRC | AM_HAL_STIMER_CFG_COMPARE_A_ENABLE);
+			     TIMER_CLKSRC | AM_HAL_STIMER_CFG_COMPARE_A_ENABLE |
+			     AM_HAL_STIMER_CFG_COMPARE_B_ENABLE);
 #endif
 	g_last_time_stamp = am_hal_stimer_counter_get();
 
 	NVIC_ClearPendingIRQ(TIMER_IRQ);
 	IRQ_CONNECT(TIMER_IRQ, 0, stimer_isr, 0, 0);
```

Comment:
On Apollo3, each STIMER comparator have separate interrupt vector. I think IRQ24 should also be connected for COMPARE_B interrupt to actually work. 
![image](https://github.com/user-attachments/assets/d84aa89e-b93c-4775-a8aa-664150ea3ae8)

![image](https://github.com/user-attachments/assets/9d12cf2e-e2cd-4cbc-8c6a-7d216f6e5d76)

Reply:
That's correct, I added one more commit to separate the two comparator interrupts, thanks for the comment